dist/clouduploader.rb

```diff
@@ -4,20 +4,86 @@
 require 'json'
 require 'ostruct'
 require 'open3'
+require 'fileutils'
 
 module CloudUploader
+  # Module method for uploading the image depending on the platform
+  def self.upload(user, platform, backend_image_file, job_data_file, image_filename, result_path)
+    STDOUT.sync = true
+    start = Time.now
+
+    job_data = JSON.parse(File.read(job_data_file))
+
+    case platform
+    when 'ec2'
+      FileUtils.ln_s(backend_image_file, File.join(Dir.pwd, image_filename)) # link file into working directory
+      EC2.upload(image_filename, job_data, result_path)
```

Comment:
[nitpick] I would prefer to have a class with attributes and not class methods like
```
EC2.new(image_filename, job_data, result_path).upload
```